ludwig/contribs/comet.py

```diff
@@ -58,13 +68,46 @@ def train(self, *args, **kwargs):
             logging.error("comet_ml.Experiment() had errors. Perhaps you need to define COMET_API_KEY")
             return
 
+        logging.info("comet.train() called......")
         cli = self._make_command_line(args)
         self.experiment.set_code(cli)
         self.experiment.set_filename("Ludwig CLI")
         self._log_html(cli)
         config = comet_ml.get_config()
         self._save_config(config)
 
+    def train_model(self, *args, **kwargs):
+        logging.info("comet.train_model() called......")
+        if self.experiment:
+            model = args[0]
+            model_definition = args[1]
+            model_definition_path = args[2]
+            if model:
+                self.experiment.set_model_graph(str(model.graph.as_graph_def()))
+            if model_definition:
+                if model_definition_path:
+                    base_name = os.path.basename(model_definition_path)
+                else:
+                    base_name = "model_definition.yaml"
+                if "." in base_name:
```

Comment:
Not 100% sure, why do you need to do this.

Reply:
Not sure either, but it appears that at this point in the process, model_definition is a JSON data structure and so we save it as json.

Reply:
I checked, in that place `model_definition` has already been parsed and merged with the defaults, so it is a nested dictionary. You are passing the full dictionary to `self.experiment.log_asset_data` so I guess what that function does behing the scenes is to serialize the dictionary into a json file, so you need to provide a json file name, but that's something your function does, not Ludwig I believe. So depending of what you want to do, you may change the `self.experiment.log_asset_data` to provide also a strategy parameter on how to serialize (maybe default is json, but the user may specify yaml if they want), or maybe you want to make explicit in the docs of that function that, if a dictionary is provided, it is going to be serialized as json. Whatever works better for you.